### PR TITLE
FR-1768 Send order not approved

### PR DESCRIPTION
**Pull request by jakub-kozlowski**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/FR-1768

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/ConsentOrderApprovedController.java [hunk 1]
```diff
@@ -145,9 +143,8 @@
                 caseData = mapper.readValue(mapper.writeValueAsString(caseData), HashMap.class);
                 caseDetails.setData(caseData);
             }
-        } else {
-            caseData.put(UPLOAD_ORDER, caseData.get(GENERAL_ORDER_LATEST_DOCUMENT));
```
Comment (shaunganley) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/ConsentOrderApprovedController.java:149]: whats the reason for removing this?

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/helper/DocumentHelper.java [hunk 2]
```diff
@@ -222,6 +230,10 @@ public String formatAddressForLetterPrinting(Map<String, Object> address) {
             .collect(Collectors.joining("\n"));
     }
 
+    public BulkPrintDocument getCaseDocumentAsBulkPrintDocument(CaseDocument caseDocument) {
```
Comment (shaunganley) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/helper/DocumentHelper.java:233]: we already have this method `getCaseDocumentAsBulkPrintDocument`

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/helper/DocumentHelper.java [hunk 3]
```diff
@@ -267,6 +279,6 @@ public static BulkPrintDocument caseDocumentToBulkPrintDocument(CaseDocument doc
     }
 
     public List<ContestedConsentOrderData> convertToContestedConsentOrderData(Object object) {
-        return (List<ContestedConsentOrderData>)object;
+        return objectMapper.convertValue(object, new TypeReference<>() {});
```
Comment (shaunganley) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/helper/DocumentHelper.java:282]: looks like you've made this method more generic? if so, can you update the method name

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/BulkPrintService.java [hunk 4]
```diff
@@ -145,7 +142,7 @@ private UUID printApplicantConsentOrderApprovedDocuments(CaseDetails caseDetails
         return bulkPrintFinancialRemedyLetterPack(caseDetails.getId(), applicantDocuments);
     }
 
-    private UUID bulkPrintFinancialRemedyLetterPack(Long caseId, List<BulkPrintDocument> documents) {
+    public UUID bulkPrintFinancialRemedyLetterPack(Long caseId, List<BulkPrintDocument> documents) {
```
Comment (shaunganley) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/BulkPrintService.java:145]: if you're making this public make sure you've added unit tests

**Review by shaunganley — CHANGES_REQUESTED** [review]
not seeing much unit test coverage being added for new code

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/ConsentOrderApprovedController.java (hunk 1 above)
Comment (jakub-kozlowski) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/ConsentOrderApprovedController.java:149]: I don't see it's required - there's logic in bulkprint methods added to send general order if there are no "contested case consented not approved" orders.

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/helper/DocumentHelper.java (hunk 2 above)
Comment (jakub-kozlowski) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/helper/DocumentHelper.java:233]: couldn't find it - GitHub link?

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/helper/DocumentHelper.java (hunk 3 above)
Comment (jakub-kozlowski) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/helper/DocumentHelper.java:282]: it wasn't about making it generic but about making it actually work - just casting an object to a list didn't work when the object was a map - using `convertValue` did the conversion from a map to a list

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/BulkPrintService.java (hunk 4 above)
Comment (jakub-kozlowski) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/BulkPrintService.java]: good spot - this was a hangover from previous refactoring no longer needed - reverted this change.

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/helper/DocumentHelper.java [hunk 5]
```diff
@@ -151,19 +156,22 @@ public CaseDetails prepareLetterToApplicantTemplateData(CaseDetails caseDetails)
...
+        String respondentName =  buildFullName(caseData,
+            isConsentedApplication ? CONSENTED_RESPONDENT_FIRST_MIDDLE_NAME : CONTESTED_RESPONDENT_FIRST_MIDDLE_NAME,
+            isConsentedApplication ? CONSENTED_RESPONDENT_LAST_NAME : CONTESTED_RESPONDENT_LAST_NAME);
 
         if (isApplicantRepresentedByASolicitor(caseData)) {
             log.info("Applicant is represented by a solicitor");
             reference = nullToEmpty((caseData.get(SOLICITOR_REFERENCE)));
-            addresseeName = nullToEmpty((caseData.get(CONSENTED_SOLICITOR_NAME)));
-            addressToSendTo = (Map) caseData.get(CONSENTED_SOLICITOR_ADDRESS);
+            addresseeName = nullToEmpty((caseData.get(isConsentedApplication ? CONSENTED_SOLICITOR_NAME : CONTESTED_SOLICITOR_NAME)));
```
Comment (shaunganley) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/helper/DocumentHelper.java:173]: can you add in a test for contested vs consented

**Review by shaunganley — APPROVED** [review]

**Review by ocarleton — APPROVED** [review]

**Pull request merged**